src/logicsig.js

```diff
@@ -16,6 +16,14 @@ class LogicSig {
         this.tag = Buffer.from("Program");
 
         assert(logic.checkProgram(program, args));
+        if (args) {
+            assert(Array.isArray(args))
+            function checkType(arg) {
+                let theType = typeof arg;
+                return ((theType == "string") || (theType == "number") || (arg.constructor == Uint8Array) || (arg.constructor == Buffer));
```

Comment:
Nit: can you use `Buffer.isBuffer` to check for Buffers? That will work slightly better in browsers.